File: Code/BeforeLegends/Assets/Scripts/Utilities/DropChance.cs
using UnityEngine;
using System.Collections;

[System.Serializable]
public class DropChance
{
    public string tileType;

    public float woodDropChance;
    public float stoneDropChance;
    public float foodDropChance;
    public float soulsDropChance;

    public float chance;

    public void overallDropChance()
    {
        chance = woodDropChance + stoneDropChance + foodDropChance + soulsDropChance;
    }

    public int returnDrop()
    {
        float rand = Random.Range(0.0f, chance);
        if (rand >= 0 && rand < woodDropChance)
            return 2;
        if (rand > woodDropChance && rand < chance - foodDropChance - soulsDropChance)
            return 1;
        if (rand > chance - foodDropChance - soulsDropChance && rand < chance - soulsDropChance)
            return 0;
        if (rand > chance - soulsDropChance && rand <= chance)
            return 3;
        return 999;
    }
}

File: Code/BeforeLegends/Assets/Scripts/Utilities/DropChance.js
//DONE


public class DropChance{
 var tileType : String;

 var woodDropChance : float;
 var stoneDropChance : float;
 var foodDropChance : float;
 var soulsDropChance : float;

 var chance : float;

 public function overallDropChance() {
 	chance = woodDropChance + stoneDropChance + foodDropChance + soulsDropChance;
 }
 public function returnDrop() : int {
 	var rand : float = Random.Range(0.0, chance);
 	if(rand >= 0 && rand < woodDropChance)
 		return 2;
 	if(rand > woodDropChance && rand < chance - foodDropChance - soulsDropChance)
 		return 1;
 	if(rand > chance - foodDropChance - soulsDropChance && rand < chance - soulsDropChance)
 		return 0;
 	if(rand > chance - soulsDropChance && rand <= chance)
 		return 3;
 }
}